kafka/consumer.go

```diff
@@ -75,16 +76,8 @@ func NewConsumer(zookeeperConnectionString string, consumerGroup string, topics
 	}, nil
 }
 
-func NewPerseverantConsumer(zookeeperConnectionString string, consumerGroup string, topics []string, config *consumergroup.Config, initialDelay time.Duration, retryInterval time.Duration) (Consumer, error) {
-	consumer := &perseverantConsumer{sync.RWMutex{}, zookeeperConnectionString, consumerGroup, topics, config, nil}
-
-	go func() {
-		if initialDelay > 0 {
-			time.Sleep(initialDelay)
-		}
-		consumer.connect(retryInterval)
-	}()
-
+func NewPerseverantConsumer(zookeeperConnectionString string, consumerGroup string, topics []string, config *consumergroup.Config, retryInterval time.Duration) (Consumer, error) {
+	consumer := &perseverantConsumer{sync.RWMutex{}, zookeeperConnectionString, consumerGroup, topics, config, nil, retryInterval}
```

Comment:
🤔 Do you need to explicitly pass a `sync.RWMutex{}` here? Or is it inferred because a `perseverantConsumer` __is__ a `sync.RWMutex{}`

Reply:
It turns out it's not required _if_ you name all the other arguments, but it is necessary otherwise.